### PR TITLE
Installation and implementation of Twig in EngineBlock

### PR DESCRIPTION
**Pull request by MKodde**
This PR is part of the changes described in [Pivotal](https://www.pivotaltracker.com/story/show/155358923)

In this PR:
* [x] Install Twig
* [x] Update the EB welcome page with a new Twig template as proof of concept.
* [x] Rewrite of all Zend View based templates to Twig.
* [x] Removal of 'cached' views from the `/app/Resources/views/*` folder
* [x] Added unit tests for some of the Twig extensions
* [x] Fully functional language switcher.
* [x] Updated UPGRADING.md and [Development guidlines wiki page](https://github.com/OpenConext/OpenConext-engineblock/wiki/Development-Guidelines)

Not in this PR:
* Support for the `transChoise`. This might be addressed in an upcoming PR.

### CODE REVIEW

File: library/EngineBlock/Application/DiContainer.php [hunk 1]
```diff
@@ -106,6 +106,14 @@ public function getMetadataRepository()
         return $this->container->get('engineblock.metadata.repository.cached_doctrine');
     }
 
+    /**
+     * @return EngineBlock_Attributes_Metadata
+     */
+    public function getMetadataDefinitions()
```
Comment (jorissteyn) [inline library/EngineBlock/Application/DiContainer.php]: Why can't we use `getAttributeMetadata`?

Comment (MKodde) [reply library/EngineBlock/Application/DiContainer.php]: Fixed this

**Comment by jorissteyn** [comment]
@tvdijen the changes we're proposing here might have an impact on how you deploy EngineBlock. Are you using a custom theme or custom templates in your EngineBlock implementation? We're curious to know if, how and why you modify templates.

**Comment by tvdijen** [comment]
I think it's fine...  I only make some changes to the .sass files to change color schemes or to hide some elements of the header / footer...

The only template I touch is this one: https://github.com/SSCICT-IAM/OpenConext-engineblock/commit/3ccce810b41929452cdcdba489dd5b5a766c5c67
I guess I have to redo that one when this PR is being released... No biggie

**Comment by jorissteyn** [comment]
Thanks that's interesting. We'll try to take into account your use-case for 8673d04(https://www.pivotaltracker.com/story/show/155467582)

File: application/configs/application.ini [hunk 2]
```diff
@@ -185,7 +185,6 @@ cookie.lang.expiry = 5184000 ; 60 days in seconds
 
 defaults.title      = "OpenConext"
 defaults.header     = "OpenConext"
```
Comment (MKodde) [inline application/configs/application.ini]: @jorissteyn, as you are the configuration specialist I come running to you for a question:

1. When overwriting the ` defaults.header` ini setting in application.ini
2. Clearing my caches
3. And running `bin/composer/dump-required-ini-params.sh`

I expect to see the updated header value in my `ini_parameters.yml` file. But the previous value is still set.

Am I overlooking something realy obvious?

Comment (MKodde) [reply application/configs/application.ini]: Okay, I overlooked something quite obvious.. My `/etc/openconext/engineblock.ini` was overriding the settings defined in my `application/configs/application.ini` file.

Comment (jorissteyn) [reply application/configs/application.ini]: As discussed: the configuration in /etc/openconext overrides the configuration in application/.

**Comment by jorissteyn** [comment]
Merging, some minor issues in the debug templates will be fixed in the Zend_Translate PR.

**Pull request merged**